### PR TITLE
[MM-37108] Fix possible nil dereference in team edition

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes a panic due to `nil` pointer dereference of `Srv().LicenseManager`.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-37108

#### Release Note

```release-note
Fixed a panic in the getPrevTrialLicense API request when loading system console on team edition.
```

#### Related PRs

https://github.com/mattermost/mattermost-webapp/pull/8406

### CODE REVIEW

File: api4/license.go [hunk 1]
```diff
@@ -178,6 +178,11 @@ func requestTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if c.App.Srv().LicenseManager == nil {
+		c.Err = model.NewAppError("requestTrialLicense", "api.license.upgrade_needed.app_error", nil, "", http.StatusBadRequest)
```
[the remark this answers is not in the document]
Comment (streamer45) [reply api4/license.go:182]: Sure. No strong opinion from my side.

**Review by agnivade — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

**Comment by amyblais** [comment]
@furqanmlk Kind reminder for PR review.

**Review by furqanmlk — APPROVED** [review]
Working as Expected.

image

image

**Comment by metanerd** [comment]
Adding the EETests label again solved the issue.

**Comment by streamer45** [comment]
> Adding the EETests label again solved the issue.

Thank you :raised_hands:

**Pull request merged**